packages/code-du-travail-modeles/src/modeles/conventions/1516_organismes_formation/salary.ts

```diff
@@ -27,28 +28,22 @@ export class ReferenceSalary1516
     salairesPendantPreavis,
   }: ReferenceSalaryProps<SupportedCcIndemniteLicenciement.IDCC1516>): number {
     const rankedSalaires = rankByMonthArrayDescFrench(salaires);
-    const rankedSalairesPendantPreavis = rankByMonthArrayDescFrench(
-      salairesPendantPreavis
-    );
     const salaryValues = rankedSalaires.map((a) => a.value).filter(nonNullable);
-
     const moyenneSalaires = sum(salaryValues) / rankedSalaires.length;
 
-    const totalSalaryValues = [
-      ...rankedSalairesPendantPreavis.map((a) => a.value).filter(nonNullable),
-      ...salaryValues,
-    ];
-
-    const meilleurSalaireDes3DerniersMois: number = Math.max(
-      ...totalSalaryValues.slice(0, 3)
+    const rankedSalairesPendantPreavis = rankByMonthArrayDescFrench(
+      salairesPendantPreavis
     );
-
-    const primesPendantPreavis = rankedSalairesPendantPreavis
-      .map((v) => v.prime)
-      .filter(nonNullable);
+    const last3salaries = [...rankedSalairesPendantPreavis, ...rankedSalaires]
+      .filter((s) => nonNullable(s?.value))
+      .slice(0, 3)
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      .sort((first, second) => second.value! - first.value!);
```

Comment:
Ce ne serait pas plus lisible avec un max avec un reduce ?
```suggestion
      .reduce((current, value) => Math.max(current, value), 0);
```

Reply:
ah yes

Reply:
je pense qu'une reduce est rarement plus lisible, je l'utilise plutôt pour optimiser les perfs en général. Perso le sort me convient bien, je trouve ca compréhensible.

Reply:
je trouve juste dommage les ! sur les value. Peut être qu'on pourrait mettre les valeurs à 0 si undefined ?